Buildkite/ephemeral-agent.md

```diff
@@ -0,0 +1,107 @@
+**NOTE**: The VM IP and VNC ports are displayed once the VM is deployed in Orka.  
+3. Verify [Homebrew][homebrew] is installed on the VM. To install Homebrew, see [here][homebrew].
+4. Install the Buildkite agent on the VM by running:  
+    ```
+    brew tap buildkite/buildkite
+    brew install buildkite-agent
+    ```
+5. Save the VM image by running `orka image save` locally.
+
+## Set up an Orka VM config
+
+To allow the wrapper agent to spin up Orka VMs, create an Orka VM config that uses the base image containing the Buildkite agent.  
+A VM config is the template used to spin up a new Orka VM.
+
+You can create up an Orka VM config using the Orka [CLI][cli] or [REST API][api]. For more information about, see the Orka [quick start guide][quick-start].
+
+## Set up a Buildkite agent
+
+To set up a Buildkite wrapper agent:
+1. Install and configure a Buildkite agent on a machine that has network visibility to your Orka environment. For more information on how to install and configure a Buildkite agent, see the [instructions][agent-instructions] provided by Buildkite.
+2. Copy the provided scripts to the agent machine: [base.sh](scripts/base.sh), [bootstrap.sh](scripts/bootstrap.sh), [run.sh](scripts/run.sh). 
+**Note**: All scripts should be in the same directory. For example you can add them to `/usr/local/var/buildkite-agent/`. Overwrite the existing `bootstrap.sh` file if asked.
+3. Verify the scripts can be executed by running `chmod +x path_to_script` in the command line.
+4. Copy the provided [pre-exit hook](scripts/pre-exit) to the agent machine. You should add it to the agent hooks directory. It can be found in the `buildkite-agent.cfg` file under the `hooks-path` property. The location of the `buildkite-agent.cfg` varies depending on your OS. For more information, see your [platform's installation article][agent-instructions].
+5. Verify the hook can be executed by running `chmod +x path_to_hook` in the command line.
+6. Verify [jq][jq] is installed. For more information, see [jq][jq] page.
+7. Verify the machine has network visibility to the Orka environment. If the machine is part of the Orka environment, skip this step. You can use any VPN client to connect to the Orka environment. For more information, see your Orka [IP Plan][ip-plan]
+
+## Buildkite environment variables
+
+The provided scripts expect the following environment variables to be set:
+
+* `ORKA_USER` - User used to connect to the Orka environment. Created by running `orka user create`
+* `ORKA_PASSWORD` - Password used to connect to the Orka environment. Created by running `orka user create`
+* `ORKA_ENDPOINT` - The Orka endpoint. By default it is `http:/10.10.10.100`
+* `ORKA_VM_NAME` - The name of the VM to be deployed. This should match the VM config created [earlier](#set-up-an-orka-vm-config)
+* `ORKA_VM_USER` - User used to SSH to the VM
+
+By default the wrapper agent and the ephemeral agent share the same Buildkite specific environment variables.  
```

Comment:
You might need to sneak in "ephemeral agent" somewhere in the sections above. Maybe "Set up an Orka VM base image for the ephemeral image" or mention ephemeral agent when you're talking about the Orka VM configs.